src/ledger.rs

```diff
@@ -170,27 +171,55 @@ mod tests {
 
     #[test]
     fn test_next_entries() {
+        use logger;
+        logger::setup();
         let id = Hash::default();
         let next_id = hash(&id);
         let keypair = KeyPair::new();
-        let tx0 = Transaction::new(&keypair, keypair.pubkey(), 1, next_id);
+        let tx_small = Transaction::new_vote(
+            &keypair,
+            Vote {
+                version: 0,
+                contact_info_version: 2,
+            },
+            next_id,
+            2,
+        );
+        let tx_large = Transaction::new(&keypair, keypair.pubkey(), 1, next_id);
+
+        let tx_small_size = serialized_size(&tx_small).unwrap();
+        let tx_large_size = serialized_size(&tx_large).unwrap();
+        assert!(tx_small_size < tx_large_size);
+        assert!(tx_large_size < PACKET_DATA_SIZE as u64);
 
         // NOTE: if Entry grows to larger than a transaction, the code below falls over
-        let threshold = (BLOB_DATA_SIZE / 256) - 1; // 256 is transaction size
+        let threshold = (BLOB_DATA_SIZE / PACKET_DATA_SIZE) - 1;
```

Comment:
PACKET_DATA_SIZE may be the wrong magic number, what you want is sizeof(Transaction)

the test also assumes that the rest of an Entry struct is smaller than a Transaction (hence the -1)

Reply:
Ideally we would use sizeof(serialized Transaction) no?  But we don’t have a way to generate that other than serializing all the different types of Transactions at runtime.  sizeof(Transaction), the in-memory representation, is more than 300 bytes.  But when serialized always must be less than 256, which I *think* is what this const represents